firewall/prompt.go

```diff
@@ -125,9 +128,35 @@ func createPrompt(ctx context.Context, conn *network.Connection, pkt packet.Pack
 
 	// If there already is a notification, just update the expiry.
 	if n != nil {
-		n.Update(expires)
-		log.Tracer(ctx).Debugf("filter: updated existing prompt notification")
-		return
+		// Get notification state
+		n.Lock()
+		state := n.State
+		n.Unlock()
+
+		// If the notification is still active, extend and return.
+		if state == notifications.Active {
+			n.Update(expires)
+			log.Tracer(ctx).Debugf("filter: updated existing prompt notification")
+			return
+		}
+
+		// The notification is not active anymore, let's check if there is an
+		// action we can perform.
+		n.Lock()
+		action := n.SelectedActionID
+		n.Unlock()
+		if action != "" {
+			switch action {
+			case allowDomainAll, allowDomainDistinct, allowIP, allowServingIP:
+				conn.Accept("permitted via prompt", profile.CfgOptionEndpointsKey)
+			default: // deny
+				conn.Deny("blocked via prompt", profile.CfgOptionEndpointsKey)
+			}
+			return nil // Do not take further action.
+		}
```

Comment:
We already handle the response around line 61 where we are waiting on `n.Response()` channel. Why do we need to check if there's an action set here as well?